xclim/indices/generic.py

```diff
@@ -354,6 +356,106 @@ def get_daily_events(
+        This function does not handle units and can't accept Quantified objects.
+    weights: sequence of floats
+        A list of weights of the same length as the window.
+        Only supported if `win_reducer` is "mean".
+    var_reducer: {'all', 'any'}
+        If the data is a list, the condition must either be fulfilled on *all*
+        or *any* variables for the period to be considered a spell.
+
+    Returns
+    -------
+    xarray.DataArray
+        Same shape as ``data``, but boolean.
+        If ``data`` was a list, this is a DataArray of the same shape as the alignment of all variables.
+    """
+    # Checks
+    if not isinstance(data, xarray.DataArray):
+        # thus a sequence
+        if np.isscalar(thresh) or len(data) != len(thresh):
+            raise ValueError(
+                "When ``data`` is given as a list, ``threshold`` must be a sequence of the same length."
+            )
+        data = xarray.concat(data, "variable")
+        thresh = xarray.DataArray(thresh, dims=("variable",))
+    if weights is not None:
+        if win_reducer != "mean":
+            raise ValueError(
+                f"Argument 'weights' is only supported if 'win_reducer' is 'mean'. Got :  {win_reducer}"
+            )
+        elif len(weights) != window:
+            raise ValueError(
+                f"Weights have a different length ({len(weights)}) than the window ({window})."
+            )
+        weights = xarray.DataArray(weights, dims=("window",))
+
+    if window == 1:  # Fast path
+        is_in_spell = compare(data, op, thresh)
+        if not np.isscalar(thresh):
+            is_in_spell = getattr(is_in_spell, var_reducer)("variable")
+    elif (win_reducer == "min" and op in [">", ">=", "ge", "gt"]) or (
+        win_reducer == "max" and op in ["`<", "<=", "le", "lt"]
+    ):
+        # Fast path for specific cases, this yields a smaller dask graph (rolling twice is expensive!)
+        # For these two cases, a day can't be part of a spell if it doesn't respect the condition itself
+        mask = compare(data, op, thresh)
+        if not np.isscalar(thresh):
+            mask = getattr(mask, var_reducer)("variable")
+        # We need to filter out the spells shorter than "window"
+        # find sequences of consecutive respected constraints
+        cs_s = rl._cumsum_reset_on_zero(mask)
+        # end of these sequences
+        cs_s = cs_s.where(mask.shift({"time": -1}, fill_value=0) == 0)
+        # propagate these end of sequences
+        # the `.where(mask>0, 0)` acts a stopper
+        is_in_spell = cs_s.where(cs_s >= window).where(mask > 0, 0).bfill("time") > 0
+    else:
+        data_pad = data.pad(time=(0, window))
+        # The spell-wise value to test
+        # For example "window_reducer='sum'", we want the sum over the minimum spell length (window) to be above the thresh
+        if weights is not None:
+            spell_value = data_pad.rolling(time=window).construct("window").dot(weights)
```

Comment:
at the moment, `weights = [0.2, 0.2, 0.4]` and `weights = [2,2,4]`, even though they have the same ratios, won't give the same answers because we don't normalize weights. Is that intended? With `sum(weights) >1` you end up multiplying your input data by this factor. Do we want this kind of behaviour? The same spells can be achieved by diving the threshold by this same factor, I think users should play with the the threshold if they want to change the spells content, not with the input by playing with weights.  So I think normalizing the weights would be better. Thoughts?

Reply:
Also, just by curiosity, do we currently have indices that would use this mode (`mean` and non-trivial `weights`?)

Reply:
Woups, indeed, we need to normalize the weights!

No we don't have official indicators using the weights, but this was a request for PC from the INSPQ. We decided not to go forward with the weights in PC anyway, but it was already coded.